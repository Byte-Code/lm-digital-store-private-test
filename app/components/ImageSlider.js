import React, { Component, PropTypes } from 'react';
import ImmutablePropTypes from 'react-immutable-proptypes';
import Slider from 'react-slick';

import Image from './Image';

const settings = {
  infinite: false,
  slidesToShow: 1,
  slidesToScroll: 1,
  speed: 100,
  arrows: false,
  dots: true,
  dotsClass: 'slickDots'
};

export default class ImageSlider extends Component {
  static propTypes = {
    imageIDList: ImmutablePropTypes.list.isRequired,
    imageOptions: PropTypes.shape({
      height: PropTypes.number,
      width: PropTypes.number
    }).isRequired,
    alt: PropTypes.string.isRequired
  }

  renderImages() {
    const { imageIDList, imageOptions, alt } = this.props;

    return imageIDList.map(imageID => (
      <div key={imageID}>
        <Image
          fixBrightColor
          imageID={imageID}
          imageOptions={imageOptions}
          alt={alt}
        />
      </div>
    ));
  }

  render() {
    return (
      <Slider {...settings}>
        {this.renderImages()}
      </Slider>
    );
  }
}
