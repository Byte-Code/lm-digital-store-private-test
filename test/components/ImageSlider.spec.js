import React from 'react';
import { fromJS } from 'immutable';
import { shallow } from 'enzyme';

import ImageSlider from '../../app/components/ImageSlider';

describe('ImageSlider', () => {
  const imageOptions = { width: 100, height: 100 };
  const alt = 'alt';

  it('should not render the slick if only one image is present', () => {
    const imageIDList = fromJS(['image0']);
    const result = shallow(
      <ImageSlider
        imageOptions={imageOptions}
        alt={alt}
        imageIDList={imageIDList}
      />
    );
    expect(result).toMatchSnapshot();
  });

  it('should render a slick carousel when multiple images are present', () => {
    const imageIDList = fromJS(['image0', 'image1', 'image2']);
    const result = shallow(
      <ImageSlider
        imageOptions={imageOptions}
        alt={alt}
        imageIDList={imageIDList}
      />
    );
    expect(result).toMatchSnapshot();
  });
});
