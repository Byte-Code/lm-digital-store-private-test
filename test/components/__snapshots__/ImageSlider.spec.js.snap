// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageSlider should not render the slick if only one image is present 1`] = `
<div>
  <div>
    <Image
      alt="alt"
      fixBrightColor={true}
      imageID="image0"
      imageOptions={
        Object {
          "height": 100,
          "width": 100,
        }
      }
    />
  </div>
</div>
`;

exports[`ImageSlider should render a slick carousel when multiple images are present 1`] = `
<Slider
  arrows={false}
  dots={true}
  dotsClass="slickDots"
  infinite={false}
  slidesToScroll={1}
  slidesToShow={1}
  speed={100}
>
  <div>
    <Image
      alt="alt"
      fixBrightColor={true}
      imageID="image0"
      imageOptions={
        Object {
          "height": 100,
          "width": 100,
        }
      }
    />
  </div>
  <div>
    <Image
      alt="alt"
      fixBrightColor={true}
      imageID="image1"
      imageOptions={
        Object {
          "height": 100,
          "width": 100,
        }
      }
    />
  </div>
  <div>
    <Image
      alt="alt"
      fixBrightColor={true}
      imageID="image2"
      imageOptions={
        Object {
          "height": 100,
          "width": 100,
        }
      }
    />
  </div>
</Slider>
`;
